### PR TITLE
run puppet with python_shell=True on windows

### PR DESCRIPTION
**Pull request by provonet**
this should fix  issue #21059

### CODE REVIEW

File: salt/modules/puppet.py [hunk 1]
```diff
@@ -158,7 +163,7 @@ def run(*args, **kwargs):
 
     puppet.kwargs.update(salt.utils.clean_kwargs(**kwargs))
 
-    ret = __salt__['cmd.run_all'](repr(puppet), python_shell=False)
+    ret = __salt__['cmd.run_all'](repr(puppet), python_shell=puppet.useshell)
```
Comment (cachedout) [inline salt/modules/puppet.py]: Given this change, is there a chance that a person can provide malicious arguments, ala: `puppet_legit_args; do_evil_thing.sh`? I'd like to have us test this first and if it's a concern, address it.

**Comment by cachedout** [comment]
Hi @provonet I left one comment in-line. There are also a few lint errors to be cleaned up. Could you take a look? I'd like to get this in though, it looks great! Thanks.

File: salt/modules/puppet.py (hunk 1 above)
Comment (cachedout) [inline salt/modules/puppet.py]: Ah, you know what? This is Windows only. I missed that before. I think this is actually fine. Ignore my previous comment.

**Comment by provonet** [comment]
The other change (implement function to read the puppet last run report) is not finished yet it was not my intention to include this in the same pull request. 
I moved this change to a seprate branch, so that that you can merge the windows fix seprate. 
so probably better that I will close this merge request and create a new one.

**Pull request closed without merging**